libs/s25main/Game.cpp

```diff
@@ -49,8 +49,8 @@ void Game::Start(bool startFromSave)
         if(ggs_.objective == GameObjective::EconomyMode)
         {
             unsigned int selection = ggs_.getSelection(AddonId::ECONOMY_MODE_GAME_LENGTH);
-            world_.econHandler = std::make_unique<EconomyModeHandler>(
-              std::chrono::minutes(AddonEconomyModeGameLengthList[selection]) / SPEED_GF_LENGTHS[referenceSpeed]);
+            world_.setEconHandler(std::make_unique<EconomyModeHandler>(
```

Comment:
should we probably  hand over the calculation to the econ-handler? What I think where we have a problem: If we pause the game, the time will still count on for this kind of goal, but I would expect that paused time does not count

Reply:
> but I would expect that paused time does not count

It does not. The EconomyModeHandler is passed a number of GFs here

Reply:
thats what I meant, so why not simply pass `AddonEconomyModeGameLengthList[selection]` inside instead of std::chrono::minutes?

Reply:
Because this selection is a gui choice and should be independent. This makes tests simpler 
And again: this does not pass minutes here, it passes GFs. Check the constructor. :)

Reply:
I moved the "minutes" into `AddonEconomyModeGameLengthList` which makes more sense and might avoid the confusion here